src/sweep_event.js

```diff
@@ -65,20 +65,24 @@ export default class SweepEvent {
     this.prevInResult = null;
 
     /**
-     * Does event belong to result?
-     * @type {Boolean}
+     * Type of result transition (0 = not in result, +1 = out-in, -1, in-out)
+     * @type {Number}
      */
-    this.inResult = false;
-
+    this.resultTransition = 0;
 
     // connection step
 
     /**
-     * @type {Boolean}
+     * @type {Number}
      */
-    this.resultInOut = false;
+    this.otherPos = -1;
 
-    this.isExteriorRing = true;
+    /**
+     * @type {Number}
+     */
+    this.outputContourId = -1;
+
+    this.isExteriorRing = true;   // TODO: Looks unused, remove?
```

Comment:
Yeah I think this is obselete

Reply:
I looked into removing it, but it is still used for determining the input contour IDs, which in turn is used in a few places across the algorithm. I have a feeling that both can be removed, but it is a bit non-trivial to do so, and can still be done in a later PR.